tests/integration/test_ec2.py

```diff
@@ -277,3 +304,76 @@ def test_describe_vpc_endpoints_with_filter(self, ec2_client):
 
         # clean up
         ec2_client.delete_vpc(VpcId=vpc_id)
+
+    @pytest.mark.aws_validated
+    @pytest.mark.parametrize("id_type", ["id", "name"])
+    def test_modify_launch_template(self, ec2_client, create_launch_template, id_type):
+        launch_template_result = create_launch_template(f"template-with-versions-{short_uid()}")
+        template = launch_template_result["LaunchTemplate"]
+
+        # call the API identifying the template wither by `LaunchTemplateId` or `LaunchTemplateName`
+        kwargs = (
+            {"LaunchTemplateId": template["LaunchTemplateId"]}
+            if (id_type == "id")
+            else {"LaunchTemplateName": template["LaunchTemplateName"]}
+        )
+
+        new_version_result = ec2_client.create_launch_template_version(
+            LaunchTemplateData={"ImageId": PUBLIC_AMAZON_UBUNTU_IMAGE}, **kwargs
+        )
+
+        new_default_version = new_version_result["LaunchTemplateVersion"]["VersionNumber"]
+        ec2_client.modify_launch_template(
+            LaunchTemplateId=template["LaunchTemplateId"],
+            DefaultVersion=str(new_default_version),
+        )
+
+        modified_template = ec2_client.describe_launch_templates(
+            LaunchTemplateIds=[template["LaunchTemplateId"]]
+        )
+        assert modified_template["LaunchTemplates"][0]["DefaultVersionNumber"] == int(
+            new_default_version
+        )
+
+
+@pytest.mark.aws_validated
+def test_raise_modify_to_invalid_default_version(ec2_client, create_launch_template):
+    launch_template_result = create_launch_template(f"my-first-launch-template-{short_uid()}")
+    template = launch_template_result["LaunchTemplate"]
+
+    with pytest.raises(ClientError) as e:
+        ec2_client.modify_launch_template(
+            LaunchTemplateId=template["LaunchTemplateId"], DefaultVersion="666"
+        )
+    assert e.value.response["ResponseMetadata"]["HTTPStatusCode"] == 400
+    assert e.value.response["Error"]["Code"] == "InvalidLaunchTemplateId.VersionNotFound"
+
+
+@pytest.mark.aws_validated
+def test_raise_when_launcTemplateData_missing(ec2_client):
+    with pytest.raises(ClientError) as e:
+        ec2_client.create_launch_template(
+            LaunchTemplateName=f"unique_name-{short_uid()}", LaunchTemplateData={}
+        )
+    assert e.value.response["ResponseMetadata"]["HTTPStatusCode"] == 400
+    assert e.value.response["Error"]["Code"] == "MissingParameter"
+
+
+@pytest.mark.aws_validated
+def test_raise_invalid_launch_template_name(ec2_client, create_launch_template):
```

Comment:
Nit: remove `ec2_client` fixture if not used

Reply:
wilco, is there a way to set up the linter to catch those?

Reply:
I don't think our linters are configured to spot this. My IDE catches this however